### DISCUSSION TITLE
Easy way to check the current page inside resource class

### DISCUSSION THREAD

**Discussion by migueliriano** [filamentphp/filament#10042] [Ideas]
Sometimes we have a different behavior when you are in an edit or creating form using a resource, so for that, we need to identify what is the page that is active at the moment.

For example, if we create a Property resource with the following pages:

- CreateProperty
- EditProperty
- ListProperties

I want to check if the current page is the `EditProperty` inside the `PropertyResource` class I have to do the next:

```php
$isEditPage = Route::currentRouteName() === Pages\EditProperty::getRouteName();
```

I propose to have a helper method inside `Filament\Resources\Pages\Page` class that is extended for all the pages that check if it's the current page:

This is the code that can be inside the `Filament\Resources\Pages\Page` class:

```php
public static function isCurrentPage() : bool
{
   return Route::currentRouteName() === static::getRouteName();
}
```

This is how to use it:

```php
Pages\CreateProperty::isCurrentPage(); // return boolean
Pages\EditProperty::isCurrentPage(); // return boolean
Pages\ListProperties::isCurrentPage(); // return boolean
```

Example use in real life: In this case, I want to have the `Wizard` component skippable only on the **edit page**:

```php
$form ->schema([
      Forms\Components\Wizard::make([...])->skippable(Pages\EditProperty::isCurrentPage())
 ]);
```

**Comment by VictorPennington** [comment]
Hi,

If you are working from a resource or from a relation manager, why not use Utilities Injections to achieve what you're trying to do, in this case by Injecting the current operation?

You then could do something like:


```
use Filament\Forms\Form;

class YourResource extends Resource
{
protected static ?string $model = YourModel::class;

public static function form(Form $form): Form
return $form ->schema([
     Forms\Components\TextInput::make('yourField')
                    ->hint(fn($operation) => "This is a ". $operation . " operation") // you can even inject multiple utilities
 ]);
}

```

And the result:

![image](https://github.com/filamentphp/filament/assets/83166965/a493c6b3-3217-4a34-a18f-36a59e27f7c7)



Check the documentation below, they are really powerful

https://filamentphp.com/docs/3.x/forms/advanced#injecting-the-current-form-operation
By the way, how do you make your code here colorful?

**Reply by migueliriano** [reply]
@VictorPennington - Thank you for that approach, I still see it as a workaround because we have to access the operation and do the logic manually, instead of having a helper that does that for us.

Anyway, this can be an alternative solution, thank you for that.

Related to the code color, just create a code block as you did above and add `php` at the beginning.

Example:

\`\`\`php 

PHP Code

\`\`\`

That can work for any other language like `css` `js` etc.

**Comment by BerendSpigt** [comment]
I was looking for a way to only show relations on the view page and not on the edit page. I did not find a "Filament" way (maybe there is), but luckily i stumbled on this post. I had to use getRouteName('admin')) because I use multiple panels.

So this is my code in the UserResource:
```php
    use App\Filament\Admin\Resources\UserResource\Pages;
    public static function getRelations(): array
    {
        if (Route::currentRouteName() !== Pages\EditUser::getRouteName('admin')) {
            return [
                MembershipsRelationManager::class,
            ];
        }
        return [];
    }
```
    
I don't think the suggested $operation injection is available here, so I think the the proposed helper function would be a useful addition.

**Comment by migueliriano** [comment]
@danharrin - What do you think about this? It looks like this can be a potentially useful addition, please let me know if we can move forward and start working on the PR (I can work on that).

**Reply by danharrin** [reply]
Already achievable using `->skippable(fn (string $operation) => $operation === 'edit')` or `->skippable(fn ($livewire) => $livewire instanceof EditProperty)` so no, sorry not interested

**Reply by migueliriano** [reply]
Are you able to use the same approach when you don't have access to the $operation injection? for example in this case on `getRelations`: https://github.com/filamentphp/filament/discussions/10042#discussioncomment-9322163

**Reply by danharrin** [reply]
`canViewForRecord()` on the relation manager controls visibility based on `$pageClass`

**Reply by BerendSpigt** [reply]
Ok, I now put the following code in canViewForRecord() in the RelationManager, and this works perfectly. 

```php
    public static function canViewForRecord(Model $ownerRecord, string $pageClass): bool
    {
        return class_basename($pageClass) === 'ViewUser';
    }
```
    
Thanks for pointing in the right direction, feels better to use the Filament way.

**Comment by jeytii** [comment]
I think this is more convenient:
```php
protected static function currentPageIs(string $name): bool
{
    $page = match ($name) {
        'index' => Pages\ListInvoices::getRouteName(/** panel name here (optional) */),
        'create' => Pages\CreateInvoice::getRouteName(/** panel name here (optional) */),
        'view' => Pages\ViewInvoice::getRouteName(/** panel name here (optional) */),
        'edit' => Pages\EditInvoice::getRouteName(/** panel name here (optional) */),
        default => null,
    };

    return Route::currentRouteName() === $page;
}
```
```php
$form->schema([
    Forms\Components\Wizard::make([...])->skippable(self::currentPageIs('edit')),
]);
```
